### PR TITLE
Extend GetContainersResources with an option to emit all containers.

### PR DESCRIPTION
**Pull request by kgolab**
Includes some minor clean-up in test of the public interface.

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go [hunk 1]
```diff
@@ -46,23 +46,30 @@ func NewProvider(calculator limitrange.LimitRangeCalculator,
...
 		recommendation := vpa_api_util.GetRecommendationForContainer(container.Name, &podRecommendation)
 		if recommendation == nil {
-			klog.V(2).Infof("no matching recommendation found for container %s", container.Name)
-			continue
+			if addAll {
+				klog.V(2).Infof("no matching recommendation found for container %s, using Pod request", container.Name)
+				resources[i].Requests = container.Resources.Requests
+			} else {
+				klog.V(2).Infof("no matching recommendation found for container %s, skipping", container.Name)
+				continue
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go:61]: We should either not continue at all or continue for both addAll=true and addAll=false (after the if addAll, else) and skip the else in line 63.

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go [hunk 2]
```diff
@@ -46,23 +46,30 @@ func NewProvider(calculator limitrange.LimitRangeCalculator,
...
 		}
-		resources[i].Requests = recommendation.Target
 		defaultLimit := core.ResourceList{}
 		if limitRange != nil {
 			defaultLimit = limitRange.Default
 		}
 		containerControlledValues := vpa_api_util.GetContainerControlledValues(container.Name, vpaResourcePolicy)
 		if containerControlledValues == vpa_types.ContainerControlledValuesRequestsAndLimits {
-			proportionalLimits, limitAnnotations := vpa_api_util.GetProportionalLimit(container.Resources.Limits, container.Resources.Requests, recommendation.Target, defaultLimit)
+			proportionalLimits, limitAnnotations := vpa_api_util.GetProportionalLimit(container.Resources.Limits, container.Resources.Requests, resources[i].Requests, defaultLimit)
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go:72]: Why this change?

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go (hunk 1 above)
Comment (kgolab) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go:61]: The idea is that addAll controls whether we emit all containers, regardless of them having a recommendation or not.
If we always continue, there is no real change in behaviour.
Or did I misunderstand you comment?

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go (hunk 2 above)
Comment (kgolab) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go:72]: So we set request back into the resources - resources[i].Requests is either recommendation.Target (as previously) or container.Resources.Requests, if addAll allows.

Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go:72]: makes sense

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go (hunk 1 above)
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go:61]: Ah sorry, I misread how big the else part is. 

Then we could do 
if !addAll {
    ...
    continue
}
so that you can drop the else

but it's not very important

**Review by bskiba — APPROVED** [review]

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go (hunk 1 above)
Comment (kgolab) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/recommendation/recommendation_provider.go:58]: Changed, I trust your golang instincts way more than mine.
